hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -857,11 +861,29 @@ public HoodieRestoreMetadata restoreToInstant(final String instantTime, boolean
         }
         return restoreMetadata;
       } else {
-        throw new HoodieRestoreException("Failed to restore " + config.getBasePath() + " to commit " + instantTime);
+        throw new HoodieRestoreException("Failed to restore " + config.getBasePath() + " to commit " + savepointToRestoreTimestamp);
       }
     } catch (Exception e) {
-      throw new HoodieRestoreException("Failed to restore to " + instantTime, e);
+      throw new HoodieRestoreException("Failed to restore to " + savepointToRestoreTimestamp, e);
+    }
+  }
+
+  private Triple<String, Option<HoodieRestorePlan>, HoodieTable<T, I, K, O>> getRestorePlan(final String savepointToRestoreTimestamp, boolean initialMetadataTableIfNecessary) throws IOException {
+    //Use the previous restore if it failed
+    HoodieTable<T, I, K, O> table = initTable(WriteOperationType.UNKNOWN, Option.empty(), initialMetadataTableIfNecessary);
```

Comment:
Option.empty() is find for this call. internally its used for upgrade flow, and MDT initialization to understand current instant in progress. incase of restore, may be won't be any. since we wouldn't have scheduled the restore yet.